packages/flutter/test/material/elevated_button_test.dart

```diff
@@ -2400,11 +2406,103 @@ void main() {
+    const Color iconColor = Color(0xFFF000FF);
+    const double iconSize = 32.0;
+    const Color disabledIconColor = Color(0xFFFFF000);
+    Widget buildButton({ bool enabled = true }) {
+      return MaterialApp(
+        home: Material(
+          child: ElevatedButton.icon(
+            style: ElevatedButton.styleFrom(
+              iconColor: iconColor,
+              iconSize: iconSize,
+              disabledIconColor: disabledIconColor,
+            ),
+            onPressed: enabled ? () {} : null,
+            icon: const Icon(Icons.add),
+            label: const Text('Button'),
+          ),
+        ),
+      );
+    }
+
+    // Test enabled button.
+    await tester.pumpWidget(buildButton());
+    expect(tester.getSize(find.byIcon(Icons.add)), const Size(iconSize, iconSize));
+    expect(iconStyle(tester, Icons.add).color, iconColor);
+
+    // Test disabled button.
+    await tester.pumpWidget(buildButton(enabled: false));
+    expect(iconStyle(tester, Icons.add).color, disabledIconColor);
+  });
+
+  // This is a test regression for https://github.com/flutter/flutter/issues/154798.
+  test('ElevatedButton.styleFrom constructor', () {
+    const Color foregroundColor = Color(0xFF000022);
+    const Color backgroundColor = Color(0xFF000011);
+    const Color disabledForegroundColor = Color(0xFF000033);
+    const Color disabledBackgroundColor = Color(0xFF000044);
+    const Color shadowColor = Color(0xFF000066);
+    const Color surfaceTintColor = Color(0xFF000077);
+    const Color iconColor = Color(0xFF000088);
+    const double iconSize = 32.0;
+    const Color disabledIconColor = Color(0xFF000099);
+    const Color overlayColor = Color(0xFF000055);
+    const double elevation = 10.0;
+    const TextStyle textStyle = TextStyle(fontSize: 21.0);
+    const EdgeInsetsGeometry padding = EdgeInsets.all(16.0);
+    const Size minimumSize = Size(88.0, 36.0);
+    const Size fixedSize = Size(100.0, 50.0);
+    const Size maximumSize = Size(120.0, 48.0);
+    const BorderSide side = BorderSide(width: 2.0, color: Color(0xFF0000AA));
+    final OutlinedBorder shape = RoundedRectangleBorder(borderRadius: BorderRadius.circular(8.0));
+    const MouseCursor enabledMouseCursor = SystemMouseCursors.click;
+    const MouseCursor disabledMouseCursor = SystemMouseCursors.basic;
+    const VisualDensity visualDensity = VisualDensity.compact;
+    const MaterialTapTargetSize tapTargetSize = MaterialTapTargetSize.shrinkWrap;
+    const Duration animationDuration = Duration(milliseconds: 200);
+    const bool enableFeedback = true;
+    const AlignmentGeometry alignment = Alignment.center;
+    const InteractiveInkFeatureFactory splashFactory = InkRipple.splashFactory;
+    Widget backgroundBuilder(BuildContext context, Set<MaterialState> states, Widget? child) {
+      return const SizedBox();
+    }
+    Widget foregroundBuilder(BuildContext context, Set<MaterialState> states, Widget? child) {
+      return const SizedBox();
+    }
```

Comment:
We should probably avoid introducing new references to deprecated names, so maybe `Set<WidgetState>` would be the ideal parameter type.